lib/services/internalEngine/index.js

```diff
@@ -83,6 +84,13 @@ function InternalEngine (kuzzle) {
         }
 
         return result;
+      })
+      .catch(error => {
+        if (error instanceof Elasticsearch.errors.NoConnections) {
```

Comment:
Since this code portion is repeated for all ES commands invoked, wouldn't it be better to store it in a function and calling it for each catch?

Something like:

```js
  .catch(error => handleError(error));
```

With:

```js
function handleError(error) {
  if (error instanceof Elasticsearch.errors.NoConnections) {
    return Promise.reject(new ServiceUnavailableError('Elasticsearch service is not connected'));
  }

  return Promise.reject(error);
}
```